java/src/main/java/ai/rapids/cudf/HostColumnVector.java

```diff
@@ -1085,9 +1120,11 @@ private void appendChildOrNull(ColumnBuilder childBuilder, Object listElement) {
       } else if (listElement instanceof BigDecimal) {
         childBuilder.append((BigDecimal) listElement);
       } else if (listElement instanceof List) {
-        childBuilder.append((List) listElement);
+        childBuilder.append((List<?>) listElement);
       } else if (listElement instanceof StructData) {
         childBuilder.append((StructData) listElement);
+      } else if (listElement instanceof byte[]) {
+        childBuilder.appendUTF8String((byte[]) listElement);
```

Comment:
Unfortunately we don't have a unique method name, like `fromUTF8Strings`, that can disambiguate.  I'm a little concerned with the blind assumption that an array of bytes must be a UTF8 string.  I suppose it's OK for now since that's the only sensible datatype mapping we have today, and in the future we could check the datatype specified in the column builder to know for sure what the user intends.

Reply:
I added another check for the type, leveraging the current `DType.STRING`. It is not the final fix but it could be better than just the data type check.
It means if you pass in a byte array with `DType.STRING`, the data should be UTF8String bytes.
Is it OK ?

Reply:
I dug a bit deeper and `childBuilder.appendUTF8String` will already assert the type is `DType.STRING` so I don't think the extra check here is really necessary.  I was mostly thinking out loud about what can we do in the future if it's ambiguous what the user's intent is with a particular Java type.  Given the type is in the child builder, we should be able to distinguish if it becomes necessary.

Reply:
updated